### PR TITLE
bump docs dependencies & remove unneeded

### PR DESCRIPTION
**Pull request by pacrob**
## What was wrong?

Docs dependencies can be bumped and unnecessary ones removed

## How was it fixed?

Bumped and removed, plus tox.ini whitelist -> allowlist, and remove all upper-bounds  while I was here.

### To-Do

[//]: # (See: https://eth-utils.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-utils/blob/master/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
         "hypothesis>=4.43.0,<5.0.0",
         "pytest>=6.2.5,<7",
         "pytest-xdist",
-        "tox==3.14.6",
+        "tox>=3.14.6,<4",
```
Comment (pacrob) [inline setup.py:13]: Needed to open up the tox version to let docs build with 3.7

Comment (kclowes) [reply setup.py:13]: Should/can we just remove the upper bound generally here?

**Review by kclowes — APPROVED** [review]
LGTM! 🤜🤛

**Comment by kclowes** [comment]
Also just occurred to me (now that you've done a bunch of them) that we should add this to the [Project Template](https://github.com/ethereum/ethereum-python-project-template/) and then in theory we can batch these sorts of changes that need to be applied across all of our libraries. Once the changes are merged to the template, I can show you how to do that for the next one if you want, or instructions are [here](https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/template.md).

File: setup.py (hunk 1 above)
Comment (pacrob) [reply setup.py]: Upper bounds gone!

**Pull request merged**